Replicator/ReplicatorOptions.hh

```diff
@@ -36,20 +39,27 @@ namespace litecore { namespace repl {
         PropertyDecryptor       propertyDecryptor       {nullptr};
         void*                   callbackContext         {nullptr};
         std::atomic<C4ReplicatorProgressLevel> progressLevel {kC4ReplProgressOverall};
-        bool                    collectionAware         {true};
 
+        bool collectionAware() const {return _collectionAware;}
+        bool isActive()        const {return _isActive;}
+        const std::unordered_map<C4CollectionSpec, size_t>& collectionSpecToIndex() const {
```

Comment:
This is debatable. IMO, _collectionSpecToIndex is an internal structure and also it is more convenient if proving the functions such as 

```
pushOf(C4CollectionSpec) -> bool / optional(bool)
pullOf(C4CollectionSpec) -> bool / optional(bool)
collectionOption(C4CollecitonSpec) -> CollectionOptions / optional(CollectionOptions)
```

Reply:
This is not a big deal. I feel a bit weird with the function named pushOf() or pullOf(). Maybe isPush() or isPull()?

Reply:
The reason I adopted weird name if because push and pull were already in Options at the time I attempted progress gradually and kept both them both in Options and Options::CollectionOptions. (Also, the collectionIndex had default 0 and I later made it required) Later, I removed Options::push/pull. Do you prefer Options::push(CollectionIndex), I can change them. 
As for the index mapping, I think mapping to index is more useful because the index has special meaning in the program. We may always get the actual CollectionOptions from the index.

Reply:
Changed accordingly,